### PR TITLE
Use Laravel schema migration command directly

### PR DESCRIPTION
**Pull request by Jellyfrog**
It's time to get rid of the old migration files :)

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Console/Commands/MigrateCommand.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+use App\Console\LnmsCommand;
+use Illuminate\Contracts\Cache\LockTimeoutException;
+use Illuminate\Support\Facades\Cache;
+use Log;
+use Schema;
+
+class MigrateCommand extends LnmsCommand
+{
+    protected $signature = 'migrate:librenms {--force : Force running even if already running}';
```
Comment (Jellyfrog) [inline app/Console/Commands/MigrateCommand.php:30]: Didn't manage to override the real command, so gave it a "work in progress" name.

Comment (Jellyfrog) [reply app/Console/Commands/MigrateCommand.php:30]: @murrant can probably solve it. But come thinking about it, wondering if this just complicates stuff for no reason?

**Comment by murrant** [comment]
What is the goal here?  Couldn't you just remove references to the old update.php script?

File: app/Console/Commands/MigrateCommand.php (hunk 1 above)
Comment (murrant) [reply app/Console/Commands/MigrateCommand.php:30]: Wouldn't schema:migrate be more appropriate.

"THE INTERNET" indicates setting $name to the command name will work, but I have no idea.

**Comment by Jellyfrog** [comment]
I think its good to get rid of old unused files in general.
But not sure if the custom migration command adds that much?

File: app/Console/Commands/MigrateCommand.php (hunk 1 above)
Comment (Jellyfrog) [reply app/Console/Commands/MigrateCommand.php:30]: overriding the command "name" itself works, but not extending it or trying to call the original one.
Think its due to how its created;

https://github.com/laravel/framework/blob/ccefc764871fe263dd20f8f7108c18180f78241b/src/Illuminate/Database/MigrationServiceProvider.php#L117-L122

**Comment by murrant** [comment]
@Jellyfrog I'm a bit dubious about removing the old migrations just yet.  They are still needed for all those stupidly old installs.

**Comment by murrant** [comment]
Sounds good.

What do you think about overriding the regular migrate command?

**Comment by Jellyfrog** [comment]
> Sounds good.
> 
> What do you think about overriding the regular migrate command?

Didnt find a way to call the underlying command then

File: app/Console/Commands/MigrateCommand.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        try {
+            if ($this->option('force-run')) {
+                optional($lock)->forceRelease();
+            }
+            optional($lock)->block(10);
+
+            // These are for the real MigrateCommand, but we hide them from the user
+            $this->addOption('--database', null, InputOption::VALUE_OPTIONAL);
+            $this->addOption('--force', null, InputOption::VALUE_OPTIONAL, '', true);
+            $this->addOption('--pretend');
```
Comment (murrant) [inline app/Console/Commands/MigrateCommand.php:61]: Might want to allow --pretend.   It helps people fix migrations that fail.

**Comment by Jellyfrog** [comment]
Latest laravel got support for running commands one at the time. Should rely on that instead.

**Review by murrant — APPROVED** [review]

**Pull request merged**